packages/mock-npm-registry/src/index.ts

```diff
@@ -0,0 +1,269 @@
+/* eslint-disable turbo/no-undeclared-env-vars */
+import childProcess, { execSync } from "node:child_process";
+import fs, { writeFile } from "node:fs/promises";
+import os from "node:os";
+import path from "node:path";
+import util from "node:util";
+import getPort from "get-port";
+import treeKill from "tree-kill";
+import { dedent } from "ts-dedent";
+import { ConfigBuilder } from "verdaccio";
+
+const debugLog = util.debuglog("mock-npm-registry");
+const repoRoot = path.resolve(__dirname, "../../..");
+
+/**
+ * Start a mock local npm registry (using verdaccio) to host local copies of packages under test.
+ *
+ * The `targetPackages` will be published to the local npm repository along with any of their local dependencies.
+ * Any non-local dependencies will continue to be pulled from the public npm registry.
+ *
+ * @param targetPackages a list of primary packages to publish to the local registry.
+ * @returns a function that stops and deletes the registry
+ */
+export async function startMockNpmRegistry(...targetPackages: string[]) {
+	// The directory where we will put the verdaccio config and database.
+	const registryPath = await fs.realpath(
+		await fs.mkdtemp(path.join(os.tmpdir(), "local-npm-registry-"))
+	);
+
+	// The port that the verdaccio on which the server will run.
+	const registryPort = await getPort();
+
+	// The packages to publish to the local npm repository.
+	const pkgs = await getPackagesToPublish(targetPackages);
+
+	const configPath = await writeVerdaccioConfig(
+		registryPath,
+		registryPort,
+		pkgs,
+		{ withUplinks: false }
+	);
+
+	console.log(
+		`Starting up local npm registry on http://localhost:${registryPort} at ${registryPath}`
+	);
+
+	let stopServer = await startVerdaccioServer(configPath);
+
+	const tempConfig = path.join(registryPath, ".npmrc");
+	await writeFile(
+		tempConfig,
+		dedent`
+			registry=http://localhost:${registryPort}
+			//localhost:${registryPort}/:_authToken=xxxx-xxxx-xxxx-xxxx
```

Comment:
Is this `authToken` actually used? I don't see us setting it on Verdaccio. Or is it meant to ensure we don't accidentally publish to NPM?